lib/msgpax/fragment.ex

```diff
@@ -0,0 +1,25 @@
+defmodule Msgpax.Fragment do
+  @moduledoc """
+  Represents a fragment of MessagePack data.
+  """
+
+  @opaque t() :: %__MODULE__{}
+
+  defstruct [:data]
+
+  def new(data) when is_binary(data) or is_list(data) do
```

Comment:
```suggestion
  @doc false
  def new(data) when is_binary(data) or is_list(data) do
```

Reply:
🤔

Reply:
Do we want the function to be used directly?